Description: fix large numbers
Author: Yadd <yadd@debian.org>
Forwarded: not-needed
Last-Update: 2021-12-16

--- a/packages/babel-core/src/transformation/normalize-file.ts
+++ b/packages/babel-core/src/transformation/normalize-file.ts
@@ -12,7 +12,7 @@
 import cloneDeep from "./util/clone-deep";
 
 const debug = buildDebug("babel:transform:file");
-const LARGE_INPUT_SOURCEMAP_THRESHOLD = 3_000_000;
+const LARGE_INPUT_SOURCEMAP_THRESHOLD = 3000000;
 
 // These regexps are copied from the convert-source-map package,
 // but without // or /* at the beginning of the comment.
--- a/packages/babel-generator/src/index.ts
+++ b/packages/babel-generator/src/index.ts
@@ -95,7 +95,7 @@
   }
 
   if (format.compact === "auto") {
-    format.compact = code.length > 500_000; // 500KB
+    format.compact = code.length > 500000; // 500KB
 
     if (format.compact) {
       console.error(
