Description: Use system gulp
Forwarded: not-needed
Reviewed-By: Yadd <yadd@debian.org>
Last-Update: 2022-07-19

--- a/Makefile
+++ b/Makefile
@@ -12,8 +12,14 @@
 SPACE := $(EMPTY) $(EMPTY)
 COMMA_SEPARATED_SOURCES = $(subst $(SPACE),$(COMMA),$(SOURCES))
 
-YARN := yarn
+YARN := yarnpkg
+ifeq (,$(filter pkg.node-babel7.yarnpkg,$(DEB_BUILD_OPTIONS) $(DEB_BUILD_PROFILES)))
+PREFIX=
+NODE := node
+else
+PREFIX=./node_modules/.bin/
 NODE := $(YARN) node
+endif
 MAKEJS := node Makefile.js
 
 
@@ -118,11 +124,11 @@
 
 
 build-no-bundle-ci: bootstrap-only
-	$(YARN) gulp build-dev
+	$(PREFIX)gulp build-dev
 	$(MAKE) build-flow-typings
 	$(MAKE) build-dist
 
-# Does not work on Windows; use "$(YARN) jest" instead
+# Does not work on Windows; use "$(PREFIX)jest" instead
 test-only:
 	BABEL_ENV=test ./scripts/test.sh
 	$(MAKE) test-clean
@@ -202,7 +208,7 @@
 new-version:
 	$(MAKE) new-version-checklist
 	git pull --rebase
-	$(YARN) release-tool version -f @babel/standalone
+	$(PREFIX)release-tool version -f @babel/standalone
 
 # NOTE: Run make new-version first
 publish:
@@ -212,7 +218,7 @@
 		exit 1; \
 	fi
 	$(MAKE) prepublish
-	$(YARN) release-tool publish
+	$(PREFIX)release-tool publish
 	$(MAKE) clean
 
 publish-test:
